From 85719919f0b8909678ea735eaa0636dfe0f53b25 Mon Sep 17 00:00:00 2001
From: Wei Li <liwei391@huawei.com>
Date: Fri, 11 Feb 2022 18:41:58 +0800
Subject: [PATCH] kabi: cleanup config entries of kabi
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 85719919f0b8909678ea735eaa0636dfe0f53b25
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4T1NF
CVE: NA

-------------------------------------------------

Move config entries of kabi to "General setup", and make
CONFIG_KABI_SIZE_ALIGN_CHECKS depending on CONFIG_KABI_RESERVE.

Signed-off-by: Wei Li <liwei391@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 Kconfig      | 15 ---------------
 init/Kconfig | 16 ++++++++++++++++
 2 files changed, 16 insertions(+), 15 deletions(-)

diff --git a/Kconfig b/Kconfig
index 9c9b5fedd43c..745bc773f567 100644
--- a/Kconfig
+++ b/Kconfig
@@ -30,18 +30,3 @@ source "lib/Kconfig"
 source "lib/Kconfig.debug"
 
 source "Documentation/Kconfig"
-
-config KABI_SIZE_ALIGN_CHECKS
-	bool "Enables more stringent kabi checks in the macros"
-	default y
-	help
-	  This option enables more stringent kabi checks. Those must be disable
-	  in case of a debug-build because they allow to change struct sizes.
-
-config KABI_RESERVE
-	bool "Enable KABI PADDING RESERVE"
-	default y
-	help
-	  This option enables KABI padding reserve.
-	  For some embedded system, KABI padding reserve may be not necessary.
-	  Disable it on demand.
diff --git a/init/Kconfig b/init/Kconfig
index b86163991bc9..4410b711f9dc 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -2086,6 +2086,22 @@ config PROFILING
 config TRACEPOINTS
 	bool
 
+config KABI_RESERVE
+	bool "Enable KABI PADDING RESERVE"
+	default y
+	help
+	  This option enables KABI padding reserve.
+	  For some embedded system, KABI padding reserve may be not necessary.
+	  Disable it on demand.
+
+config KABI_SIZE_ALIGN_CHECKS
+	bool "Enables more stringent kabi checks in the macros"
+	default y
+	depends on KABI_RESERVE
+	help
+	  This option enables more stringent kabi checks. Those must be disable
+	  in case of a debug-build because they allow to change struct sizes.
+
 endmenu		# General setup
 
 source "arch/Kconfig"
-- 
2.26.2

